modules/andes-core/broker/src/main/java/org/wso2/andes/server/handler/ChannelFlowHandler.java

```diff
@@ -58,29 +59,35 @@ public void methodReceived(AMQStateManager stateManager, final ChannelFlowBody b
 
         channel.setSuspended(!body.getActive());
 
-        QpidAndesBridge.notifyChannelFlow(channel.getId(), body.getActive(), new DisruptorEventCallback() {
-            @Override
-            public void execute() {
-                MethodRegistry methodRegistry = session.getMethodRegistry();
-                AMQMethodBody responseBody = methodRegistry.createChannelFlowOkBody(body.getActive());
-                session.writeFrame(responseBody.generateFrame(channelId));
-            }
+        try {
+            QpidAndesBridge.notifyChannelFlow(channel.getId(), body.getActive(), new DisruptorEventCallback() {
+                @Override
+                public void execute() {
+                    MethodRegistry methodRegistry = session.getMethodRegistry();
+                    AMQMethodBody responseBody = methodRegistry.createChannelFlowOkBody(body.getActive());
+                    session.writeFrame(responseBody.generateFrame(channelId));
+                }
+
+                @Override
+                public void onException(Exception exception) {
+                    logger.error("Exception occurred while processing channel flow: " + body.getActive(),
+                                 exception);
+                    AMQChannelException channelException = body.getChannelException(AMQConstant.INTERNAL_ERROR,
+                                                                                    exception.getMessage());
+                    session.writeFrame(channelException.getCloseFrame(channelId));
+                    try {
+                        session.closeChannel(channelId);
+                    } catch (AMQException e) {
+                        logger.error("Couldn't close channel (channel id ) " + channelId, e);
+                    }
 
-            @Override
-            public void onException(Exception exception) {
-                logger.error("Exception occurred while processing channel flow: " + body.getActive(),
-                             exception);
-                AMQChannelException channelException = body.getChannelException(AMQConstant.INTERNAL_ERROR,
-                                                                                exception.getMessage());
-                session.writeFrame(channelException.getCloseFrame(channelId));
-                try {
-                    session.closeChannel(channelId);
-                } catch (AMQException e) {
-                    logger.error("Couldn't close channel (channel id ) " + channelId, e);
                 }
+            });
+        } catch (AndesException e) {
```

Comment:
Shouldn't this go inside onException? I mean if an exception is thrown while QpidAndesBridge.notifyChannelFlow() is executed, don't we need to send a channel close frame?

Reply:
OnException is triggered if you get an error while processing the event. This catch statement is to catch exceptions thrown while publishing the event. Additionally, when the exception is received the protocol engine will initiate the channel close. So we do not have to do anything.